skimage/segmentation/_join.py

```diff
@@ -121,24 +121,21 @@ def relabel_sequential(label_field, offset=1):
         raise ValueError("Offset must be strictly positive.")
     if np.min(label_field) < 0:
         raise ValueError("Cannot relabel array that contains negative values.")
-    m = label_field.max()
+    max_label = label_field.max()
     if not np.issubdtype(label_field.dtype, np.integer):
-        new_type = np.min_scalar_type(int(m))
+        new_type = np.min_scalar_type(int(max_label))
         label_field = label_field.astype(new_type)
-        m = m.astype(new_type)  # Ensures m is an integer
+        max_label = max_label.astype(new_type)  # Ensures max_label is an integer
     labels = np.unique(label_field)
     labels0 = labels[labels != 0]
-    required_type = np.min_scalar_type(offset + len(labels0))
+    new_max_label = offset - 1 + len(labels0)
+    new_labels0 = np.arange(offset, int(new_max_label + 1))
+    required_type = np.min_scalar_type(new_max_label)
```

Comment:
I also changed another subtlety regarding `required_type`.
(It is amazing how much complexity can be in such a small function.)

Reply:
>  (It is amazing how much complexity can be in such a small function.)

I know! Thank you again for your persistence here! It's a much better function thanks to your contributions!